crates/unftp-sbe-gcs/src/lib.rs

```diff
@@ -87,6 +87,7 @@ use response_body::{Item, ResponseBody};
 use std::{
     fmt::Debug,
     path::{Path, PathBuf},
+    sync::{Arc, RwLock},
```

Comment:
You will need to use the async ready tokio version of RwLock: https://docs.rs/tokio/0.2.22/tokio/sync/struct.RwLock.html

We've had deadlocks before when using the std sync one. There are exceptions as explained in the tokio docs but if in doubt use the async ones.

Reply:
Ah, good to know, I'll update